### PR TITLE
Add support dspi for s32z270dc2

### PR DESCRIPTION
**Pull request by tunguyen4585**
This PR to add support dspi for s32z27x devices
Tested on tests\drivers\spi\spi_loopback\drivers.spi.s32z_dspi.loopback
------ TESTSUITE SUMMARY START ------
SUITE PASS - 100.00% [spi_loopback]: pass = 1, fail = 0, skip = 0, total = 1 duration = 0.366 seconds
PASS - [spi_loopback.test_spi_loopback] duration = 0.366 seconds
------ TESTSUITE SUMMARY END ------

### CODE REVIEW

File: s32/mcux/devices/S32Z270/S32Z270_device.h [hunk 1]
```diff
@@ -1885,4 +1885,849 @@ typedef struct {
...
+
+/*!
+ * @addtogroup DSPI_Register_Masks DSPI Register Masks
+ * @{
+ */
+
+/*! @name MCR - Module Configuration Register */
+/*! @{ */
+
+#define SPI_MCR_HALT_MASK                          DSPI_MCR_HALT_MASK
```
Comment (manuargue) [inline s32/mcux/devices/S32Z270/S32Z270_device.h:1956]: same comment as before, can we avoid reformatting the file after copied over from mcux-sdk?

**Review by manuargue — CHANGES_REQUESTED** [review]

File: s32/mcux/devices/S32Z270/fsl_device_registers.h [hunk 2]
```diff
@@ -16,7 +16,10 @@
 #if defined(CPU_S32Z270)
 
 /* CMSIS-style register definitions */
-#include "S32Z2.h"
+#include "S32Z2_PIT.h"
```
Comment (manuargue) [inline s32/mcux/devices/S32Z270/fsl_device_registers.h:19]: >add essential header files to build with mcux

elaborate pls, why this is needed?

File: dts/nxp/s32/S32Z27-BGA594-pinctrl.h [hunk 3]
```diff
@@ -612,6 +612,9 @@
 #define PA11_DSPI_10_PCS0_O             NXP_S32_PINMUX(0, 0, 11, 1, 0, 0)
 #define PA12_DSPI_10_PCS1               NXP_S32_PINMUX(0, 0, 12, 1, 0, 0)
 #define PB6_DSPI_10_PCS2                NXP_S32_PINMUX(0, 0, 22, 6, 0, 0)
+#define DSPI_10_SIN                     NXP_S32_PINMUX(0, 0, 500, 0, 0, 0)
```
Comment (manuargue) [inline dts/nxp/s32/S32Z27-BGA594-pinctrl.h:615]: why there are no prefix for the port/pin on these three pinmuxes?

**Review by manuargue — CHANGES_REQUESTED** [review]

File: s32/mcux/devices/S32Z270/S32Z270_device.h (hunk 1 above)
Comment (manuargue) [reply s32/mcux/devices/S32Z270/S32Z270_device.h:1956]: ignore this

File: s32/mcux/devices/S32Z270/fsl_device_registers.h (hunk 2 above)
Comment (tunguyen4585) [reply s32/mcux/devices/S32Z270/fsl_device_registers.h:19]: + In _"S32Z2.h"_ included _"S32Z2_SPI.h"_ that has the struct **SPI_Type** and the macros as same as which I has defined in  _modules\hal\nxp\s32\mcux\devices\S32Z270\S32Z270_device.h_ and **mcux dspi** is using, leads to raise error re-defined. If we want to build DSPI with mcux, we cannot include "S32Z2_SPI.h"

+ If we include <S32Z2.h>, we could add unnecessary files to build

That why I want to include the essential files

I propose 2 solutions:
+ do like which implemented
+ jus remove or guard _#include "S32Z2_SPI.h"_ in _modules\hal\nxp\s32\drivers\s32ze\BaseNXP\header\S32Z2.h_ and keep them all

File: dts/nxp/s32/S32Z27-BGA594-pinctrl.h (hunk 3 above)
Comment (tunguyen4585) [reply dts/nxp/s32/S32Z27-BGA594-pinctrl.h:615]: those pin belongs to LVDS pins and quite special, 
I see that we named the macros by "port" "_" "function"
however, I see in the IOMUX sheet the port is same as function :). Should we keep the same which i implemented?

File: s32/mcux/devices/S32Z270/S32Z270_device.h [hunk 4]
```diff
@@ -1885,4 +1885,849 @@ typedef struct {
...
+#define SPI_CTAR_FMSZ(x)                           DSPI_CTAR_FMSZ(x)
+
+#define SPI_CTAR_DBR_MASK                          DSPI_CTAR_DBR_MASK
+#define SPI_CTAR_DBR_SHIFT                         DSPI_CTAR_DBR_SHIFT
+#define SPI_CTAR_DBR_WIDTH                         DSPI_CTAR_DBR_WIDTH
+#define SPI_CTAR_DBR(x)                            DSPI_CTAR_DBR(x)
+/*! @} */
+
+/*! @name CTAR_SLAVE - Clock and Transfer Attributes Register (In Slave Mode) */
+#define SPI_CTAR_SLAVE_CPHA_MASK                   (0x2000000U)
```
Comment (Dat-NguyenDuy) [inline s32/mcux/devices/S32Z270/S32Z270_device.h:2121]: I thought RTD already defines all macros we need? isn't it

**Review by Dat-NguyenDuy — CHANGES_REQUESTED** [review]
As discussed with @tunguyen4585, we need to patch mcux-sdk DSPI driver, blocking this PR until the patch is merged and picked up into this PR

File: s32/mcux/devices/S32Z270/S32Z270_device.h (hunk 4 above)
Comment (tunguyen4585) [reply s32/mcux/devices/S32Z270/S32Z270_device.h]: I remove all the RTD unsupported macros

**Comment by tunguyen4585** [comment]
> As discussed with @tunguyen4585, we need to patch mcux-sdk DSPI driver, blocking this PR until the patch is merged and picked up into this PR

The PR on mcux-sdk was merged, wait it sync to github

**Comment by Dat-NguyenDuy** [comment]
> As discussed with @tunguyen4585, we need to patch mcux-sdk DSPI driver, blocking this PR until the patch is merged and picked up into this PR

You'll need to update Patch List from [README](https://github.com/zephyrproject-rtos/hal_nxp/blob/master/mcux/README#L67) as well

**Comment by tunguyen4585** [comment]
> > As discussed with @tunguyen4585, we need to patch mcux-sdk DSPI driver, blocking this PR until the patch is merged and picked up into this PR
> 
> You'll need to update Patch List from [README](https://github.com/zephyrproject-rtos/hal_nxp/blob/master/mcux/README#L67) as well

I updated

**Review by Dat-NguyenDuy — APPROVED** [review]

**Review by manuargue — CHANGES_REQUESTED** [review]
block until zephyr pr approved

File: dts/nxp/s32/S32Z27-BGA594-pinctrl.h (hunk 3 above)
Comment (manuargue) [reply dts/nxp/s32/S32Z27-BGA594-pinctrl.h:615]: I would suggest to use to use `LVDS_` as prefix for the voltage differential pads, to indicate that they are not targetting single-ended GPIO pads that belong to a port/pin pair

Comment (tunguyen4585) [reply dts/nxp/s32/S32Z27-BGA594-pinctrl.h]: I updated it, thank

File: s32/mcux/devices/S32Z270/fsl_device_registers.h (hunk 2 above)
Comment (manuargue) [reply s32/mcux/devices/S32Z270/fsl_device_registers.h]: I see you have changed the approach in https://github.com/zephyrproject-rtos/hal_nxp/pull/435/commits/83f04dc0fdf09bc3e0b92bccee2810c039043cb1 
patches to the HAL must be documented in the s32/README

Comment (tunguyen4585) [reply s32/mcux/devices/S32Z270/fsl_device_registers.h]: I updated

Comment (manuargue) [reply s32/mcux/devices/S32Z270/fsl_device_registers.h]: I must say that I prefer the previous approach, including only what's necessary in zephyr's soc.h, but we can do it in a separate pr that address all drivers.

Comment (tunguyen4585) [reply s32/mcux/devices/S32Z270/fsl_device_registers.h]: yes, we will update in the future PRs.

**Review by manuargue — APPROVED** [review]

**Review by Dat-NguyenDuy — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Comment by manuargue** [comment]
@tunguyen4585 please rebase

**Review by manuargue — APPROVED** [review]

**Review by Dat-NguyenDuy — APPROVED** [review]

**Comment by manuargue** [comment]
zephyr pr approved by assignee, merging

**Pull request merged**